src/lib/dnssd/IncrementalResolve.cpp

```diff
@@ -343,14 +345,15 @@ CHIP_ERROR IncrementalResolver::OnIpAddress(Inet::InterfaceId interface, const I
 
 CHIP_ERROR IncrementalResolver::Take(DiscoveredNodeData & outputData)
 {
-    VerifyOrReturnError(IsActiveBrowseParse(), CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrReturnError(IsActiveCommissionParse(), CHIP_ERROR_INCORRECT_STATE);
 
     IPAddressSorter::Sort(mCommonResolutionData.ipAddress, mCommonResolutionData.numIPs, mCommonResolutionData.interfaceId);
 
-    outputData.Set<CommissionNodeData>();
-    CommissionNodeData & nodeData         = outputData.Get<CommissionNodeData>();
-    nodeData                              = mSpecificResolutionData.Get<CommissionNodeData>();
-    CommonResolutionData & resolutionData = nodeData;
+    // Set nodeData with data specific to CommissionNodeData
+    outputData.Set<CommissionNodeData>(mSpecificResolutionData.Get<CommissionNodeData>());
+
+    // set the common resolution data to nodeData
```

Comment:
I don't understand this comment.  What is `nodeData` here?

In general, this "assign to the same struct twice" needs some clear documentation about the assumptions and ordering here.  

As a followup, the fact that we are keeping `mCommonResolutionData` separately from `mSpecificResolutionData.Get<CommissionNodeData>()` which _also_ holds the common resolution data is pretty confusing and should be fixed.  But not in this PR.

Reply:
Added more descriptive comment to explain the changes done.

Reply:
Much better, thank you!